client/consensus/manual-seal/src/rpc.rs

```diff
@@ -126,6 +161,61 @@ impl<Hash: Send + 'static> ManualSealApiServer<Hash> for ManualSeal<Hash> {
 		sink.send(command).await?;
 		receiver.await.map(|_| true).map_err(|e| JsonRpseeError::to_call_error(e))
 	}
+
+	async fn forward_blocks_to(
+		&self,
+		height: <<Block as BlockT>::Header as Header>::Number,
+	) -> RpcResult<()> {
+		let best_number = self.client.info().best_number;
+		if height <= best_number {
+			return Err(JsonRpseeError::Custom(
+				"Target height is lower than current best height".into(),
+			))
+		}
+
+		let diff = height - best_number;
+		let to_height = (0..diff.saturated_into::<u64>())
+			.into_iter()
+			.map(|_| EngineCommand::SealNewBlock {
+				create_empty: true,
+				finalize: false,
+				parent_hash: None,
+				sender: None,
+			})
+			.collect::<Vec<EngineCommand<Block::Hash>>>();
+
+		let mut forward_blocks_stream = stream::iter(to_height).map(Ok);
+
+		let mut sink = self.import_block_channel.clone();
+		sink.send_all(&mut forward_blocks_stream).await?;
+
+		Ok(())
+	}
+
+	async fn revert_blocks_to(
+		&self,
+		height: <<Block as BlockT>::Header as Header>::Number,
+	) -> RpcResult<()> {
+		let best_number = self.client.info().best_number;
+		if height >= best_number {
+			return Err(JsonRpseeError::Custom(
+				"Target height is higher than current best height".into(),
+			))
+		}
+
+		let diff = best_number - height;
+
+		println!("Diff: {:?}", diff);
+
+		let reverted = self
+			.backend
+			.revert(diff, true)
```

Comment:
Backend revert method doesn't work when blocks are finalized for some reason, even though it is described that providing `true` as the second argument can revert finalized blocks (in unsafe way).